cmd/main.go

```diff
@@ -476,6 +495,20 @@ func main() {
 	}
 }
 
+func generateAPMTransactionName(request *http.Request) string {
+	route := middleware.MatchedRouteFrom(request)
+
+	if route == nil {
+		// Use the actual URL path if no route
+		// matched this request. This will make
+		// sure we can, granuarly, introspect
+		// non-grouped requests.
+		return request.URL.Path
+	}
+
+	return route.PathPattern
```

Comment:
I think we should return route.Method, just like we do in https://github.com/openshift/assisted-service/blob/3d3b762caf576bafd20a62d6e50824948e0cc238/internal/metrics/reporter.go#L34

Reply:
Is it needed so that we can track the http method too? APM already tracks the request method. This function is only responsible for generating the transaction name and APM takes care of the rest.

Reply:
This changes the transaction name to be more understandable
instead of the endponit path, the name will be "cluster_create" for example

Reply:
> This changes the transaction name to be more understandable
instead of the endponit path, the name will be "cluster_create" for example

A-ha, ok! I thought `Method` in this case was referring to the HTTP method. Instead, it is the actual action name. Let me change this, thanks for the suggestion :smiley:

Reply:
rebased and addressed this comment!